iree/compiler/Dialect/Util/IR/UtilOps.td

```diff
@@ -176,6 +176,47 @@ def Util_RangeExtentsOp : Util_PureOp<"range.extents", [
   let hasCanonicalizer = 1;
 }
 
+//===----------------------------------------------------------------------===//
+// Alignment Arithmetic
+//===----------------------------------------------------------------------===//
+
+def Util_AlignOp : Util_PureOp<"align", [
+  AllTypesMatch<["value", "result"]>
+  ]> {
+  let summary = "Aligns input values";
+  let description = [{
+     Aligns |value| up to the given power-of-two |alignment| if required.
+  }];
+
+  let arguments = (ins
+    SignlessIntegerLike:$value,
+    Index:$alignment
+  );
+
+  let results = (outs
+    SignlessIntegerLike:$result
+  );
+
+  let assemblyFormat = [{
+    $value `,` $alignment attr-dict `:` type($result)
+  }];
+
+  let builders = [
+    OpBuilder<(ins
+      "Type":$type,
+      "Value":$value,
+      "int64_t":$alignment
+    ),
+    [{
+      build($_builder, $_state, type, value,
+       $_builder.createOrFold<arith::ConstantIndexOp>($_state.location, alignment));
```

Comment:
can you shorten this further using `value.getType()`? like this one: https://github.com/google/iree/blob/e3b53df58dfd6a78dc78d6605792f94b8192de7c/iree/compiler/Dialect/Util/IR/UtilOps.td#L215-L217